Added DNS module to NanoWallet

aenima86 opened the pull request:
Implementation of DNS module in NanoWallet.

Here is an image series of the implementation: https://imgur.com/a/3FARbnD

And a short video: https://www.useloom.com/share/6c87680d596a4249984a5d5c8a3cca0d

aenima86 committed:
Add files via upload

aenima86 committed:
Add files via upload

aenima86 committed:
Add files via upload

aenima86 committed:
Add files via upload

QuantumMechanics commented:
Hi,

Very nice :)

Please can you fix the following before merge:

- `getPointerAccount` function in `search` module controller has unreachable code after return statement.
- After clicking `GO` it should open in a new tab because new popup window is blocked by Firefox
- You use a sha256 function in both controllers while you could simply use `nem.crypto.js.SHA256`. After testing it seems that it produce different hashes, means that the sha256 function you use is probably broken.
- In `configure dns` when you clean the form you should clean the json and block the send button until user has completed minimum of the form. Right now if you choose a namespace, fill the form and then change namespace it will clear the form but not the json and user can still send, which can be confusing.
- Tab title in config of `configure dns` is still "My Nano Wallet Module"
- Seems Ukrainian is not in your `languages/index.js` file

aenima86 committed:
Update nl.js

aenima86 committed:
Update index.js

aenima86 committed:
Add files via upload

aenima86 commented:
@QuantumMechanics > **Thank you for reviewing my module.**

**I have updated the code accordingly to your suggestions:**

•	getPointerAccount function in search module controller has unreachable code after return statement.
`•	FIX: the code dose not return this error anymore`

•	After clicking GO it should open in a new tab because new popup window is blocked by Firefox
`•	FIX: Firefox and Chrome will by default block popups but I added a check and if blocked the user is notified and asked to enable popups.`

•	You use a sha256 function in both controllers while you could simply use nem.crypto.js.SHA256. After testing it seems that it produce different hashes, means that the sha256 function you use is probably broken.
`•	FIX: the code is now using the nem.crypto.js.SHA256 function`

•	In configure dns when you clean the form you should clean the json and block the send button until user has completed minimum of the form. Right now if you choose a namespace, fill the form and then change namespace it will clear the form but not the json and user can still send, which can be confusing.
`•	FIX: the json is now cleared when switching namespace and the send button is blocked if json==””`

•	Tab title in config of configure dns is still "My Nano Wallet Module"
`•	FIX: title is added
`

•	Seems Ukrainian is not in your languages/index.js file
`•	FIX: the lang. file is now updated.
`
